front-end/src/molecules/SidebarWorkspace.js

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from "react";
+
+import { AddWorkspaceIcon } from "../atoms/icons";
+import SidebarWorkspaceItem from "../atoms/SidebarWorkspaceItem";
+import WorkspaceSettings from "../organisms/WorkspaceSettings";
+
+import "./SidebarWorkspace.css";
+
+const componentName = "SidebarWorkspace";
+function SidebarWorkspace( ) {  
+
+  const [ showAddWorkspace, setShowAddWorkspace ] = useState(false);
+  const [ workspaceList, setWorkspaceList ] = useState([]);
+  const [ showWorkspaceSettingPopup, setShowWorkspaceSettingPopup ] = useState(false);
+
+  useEffect(() => {
+    
+    // Do GET request here
+    
+    const fakeWorkspace = [
+      {name: "My Workspace", id: 1},
+      {name: "eThis is a very long workspace name", id: 2},
+      {name: "workspace1", id: 3},
+    ];
+
+    setWorkspaceList(fakeWorkspace);
+
+  }, []); // empty second param in useEffect will only run once (on first load)
+  
+  useEffect(() => { 
+
+  }, [showWorkspaceSettingPopup]);
+
+  const handleWorkspaceSubmit = (e) => {
+    if (e.key === "Enter") {
+      e.preventDefault();
+
+      const newWorkspaceList = [...workspaceList, {id: Math.random(), name:e.target.value}];
```

Comment:
Add the previous workspaces to the end so the new entry appears first

Reply:
done